### PR TITLE
Bl 1000 fe graph ds endpoint update

### PR DESCRIPTION
**Pull request by JamesFincher**
## Description

-Added state for the incoming object. This allows some flexibility in the future if we need different graphs.
-Added axios to fetch and store state
-Random ipsum is now the new description of the graph we received from the new graph object.

[BL-1000](https://bloomtechlabs.atlassian.net/browse/BL-1000?atlOrigin=eyJpIjoiMjZhN2Y1MmM0YTMxNDE5Zjk2MjVlMjgwM2NiOWFiYjYiLCJwIjoiaiJ9)

## Loom Video

[LOOM](https://www.loom.com/share/4fc69b5eab424fb5a0d8eed30121d796)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: src/components/pages/Analytics/TechStackGraph.jsx [hunk 1]
```diff
@@ -1,29 +1,26 @@
...
   useEffect(() => {
-    const getTechStackData = async () => {
-      await embed('#vis', url);
-    };
-    getTechStackData();
-  }, [url]);
+    axios.get(`${url}`).then(function (res) {
+      setGraphData(res.data);
+    });
+  }, url);
```
Comment (roxannebobby) [inline src/components/pages/Analytics/TechStackGraph.jsx:17]: should the dependency be in an array?

**Review by roxannebobby — APPROVED** [review]
Looks like great work only the one lint issue. The dependency may need to be an array to ensure that it only gets loaded once.

File: src/components/pages/Analytics/TechStackGraph.jsx [hunk 2]
```diff
@@ -1,29 +1,26 @@
...
 import embed from 'vega-embed';
+//vega-embed allows us to take a graph object, and display it on the ui.
+// for more information visit https://github.com/vega/vega-embed
+
 export default function TechStackGraph() {
+  const [graphData, setGraphData] = useState({ graph: {}, description: '' });
   // let url = `${process.env.REACT_APP_DS_API_URL}/graph/tech-stack-by-role`;
-  let url =
+  const url =
     'http://underdog-devs-ds-a-dev.us-east-1.elasticbeanstalk.com/graph/tech-stack-by-role';
```
Comment (tim-sansone) [inline src/components/pages/Analytics/TechStackGraph.jsx:11]: I think we should not hardcode the url here but instead pull it from our environment variables. It looks like someone had done that earlier in that commented out code on line 9.

File: src/components/pages/Analytics/TechStackGraph.jsx [hunk 3]
```diff
@@ -1,29 +1,26 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
 import embed from 'vega-embed';
+//vega-embed allows us to take a graph object, and display it on the ui.
+// for more information visit https://github.com/vega/vega-embed
+
 export default function TechStackGraph() {
+  const [graphData, setGraphData] = useState({ graph: {}, description: '' });
   // let url = `${process.env.REACT_APP_DS_API_URL}/graph/tech-stack-by-role`;
```
Comment (tim-sansone) [inline src/components/pages/Analytics/TechStackGraph.jsx:9]: Do we need to leave this commented out code here?

**Review by tim-sansone — COMMENTED** [review]
Overall good work refactoring this component to be compatible with the ds graphs endpoint!

File: src/components/pages/Analytics/TechStackGraph.jsx (hunk 2 above)
Comment (JamesFincher) [reply src/components/pages/Analytics/TechStackGraph.jsx]: I have fixed this in [this commit](https://github.com/BloomTech-Labs/underdog-devs-fe-a/pull/325/commits/27335b4e8b355f75fda938537420a710e8d39534)

Thank you for your suggestion!

File: src/components/pages/Analytics/TechStackGraph.jsx (hunk 3 above)
Comment (JamesFincher) [reply src/components/pages/Analytics/TechStackGraph.jsx]: I have fixed this in [this commit](https://github.com/BloomTech-Labs/underdog-devs-fe-a/pull/325/commits/27335b4e8b355f75fda938537420a710e8d39534)

Thank you for your suggestion!

File: src/components/pages/Analytics/TechStackGraph.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (JamesFincher) [reply src/components/pages/Analytics/TechStackGraph.jsx]: I have fixed this using an empty array as suggested [here.](https://github.com/BloomTech-Labs/underdog-devs-fe-a/pull/325/commits/e30529a6e73edf1f428405931f3a08974069dc81).

Thanks for the feedback!

**Review by tim-sansone — COMMENTED** [review]
Good job refactoring and responding to feedback. This component looks good. Great work!

**Review by zoebrantley — APPROVED** [review]
Your code looks great! Very nicely done.

**Comment by JamesFincher** [comment]
> Your code looks great! Very nicely done.

Thank you!

File: src/components/pages/Analytics/TechStackGraph.jsx [hunk 4]
```diff
@@ -1,33 +1,26 @@
 import axios from 'axios';
-import React, { useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
 import embed from 'vega-embed';
+//vega-embed allows us to take a graph object, and display it on the ui.
+// for more information visit https://github.com/vega/vega-embed
+
 export default function TechStackGraph() {
+  const [graphData, setGraphData] = useState({ graph: {}, description: '' });
```
Comment (joanterm) [inline src/components/pages/Analytics/TechStackGraph.jsx:7]: Great job adding state here and replacing lorem ipsum later on. Well done!

**Review by tim-sansone — APPROVED** [review]
Just the 1 comment on the env variables. Otherwise we're looking good!

**Pull request merged**